src/Microsoft.DotNet.Interactive.ExtensionLab/MsSqlServiceClient.cs

```diff
@@ -15,11 +15,22 @@ namespace Microsoft.DotNet.Interactive.ExtensionLab
 {
     public class MsSqlServiceClient : IDisposable
     {
-        public static readonly MsSqlServiceClient Instance = new MsSqlServiceClient();
-
         private Process _process;
         private JsonRpc _rpc;
         private bool _initialized = false;
+        private readonly string _serviceExePath;
+
+        public const string SqlToolsServiceEnvironmentVariableName = "DOTNET_SQLTOOLSSERVICE";
+
+        public MsSqlServiceClient(string serviceExePath = null)
+        {
+            _serviceExePath = serviceExePath;
+
+            if (string.IsNullOrWhiteSpace(_serviceExePath))
+            {
+                _serviceExePath = Environment.GetEnvironmentVariable(SqlToolsServiceEnvironmentVariableName);
```

Comment:
If we're unable to find a value for this, it probably deserves an exception at this point.

Reply:
Fixed by moving the check from when we start the process to the constructor.